server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -256,6 +262,14 @@ public DiscoveryNodes nodesIfRecovered() {
         return blocks.hasGlobalBlock(STATE_NOT_RECOVERED_BLOCK) ? DiscoveryNodes.EMPTY_NODES : nodes;
     }
 
+    public Map<String, TransportVersion> transportVersions() {
+        return this.transportVersions;
+    }
+
+    public TransportVersion getMinTransportVersion() {
+        return this.transportVersions.values().stream().min(Comparator.naturalOrder()).orElse(TransportVersion.CURRENT);
```

Comment:
Re calculating this on every call seems unnecessary. At minimum let's calculate this in the ctor, and also pass it along in the serialized state?

Reply:
Added a calculated field to the state